iOS/DuckDuckGo/AutoClear.swift

```diff
@@ -102,17 +103,18 @@ final class AutoClear: AutoClearing {
         }
     }
 
-    @MainActor
-    func clearDataIfEnabledAndTimeExpired(baseTimeInterval: TimeInterval = Date().timeIntervalSince1970,
-                                          applicationState: DataStoreWarmup.ApplicationState) async {
-        guard isClearingEnabled,
-            let timestamp = timestamp,
-            shouldClearData(elapsedTime: baseTimeInterval - timestamp) else { return }
+    var shouldClearDataIfTimeExpired: Bool {
+        guard isClearingEnabled, let timestamp = timestamp else { return false }
+        return shouldClearData(elapsedTime: Date().timeIntervalSince1970 - timestamp)
+    }
 
-        self.timestamp = nil
+    @MainActor
+    func clearDataDueToTimeExpired(applicationState: DataStoreWarmup.ApplicationState) async {
```

Comment:
I'm not sure I like this - the logic of using timestamp was encapsulated in previous implementation, now it leaks outside of the class. It is also possible to start clearing at wrong time. I'm thinking of leaving old `clearDataIfEnabledAndTimeExpired` as is, and just have this new method to check if time has expired.